theano/gpuarray/opt.py

```diff
@@ -210,6 +210,52 @@ def apply(self, fgraph):
                     0, 'fast_run', 'fast_compile', 'merge')
 
 
+class GraphToGPU(Optimizer):
+    """
+    Transfer the graph as a whole to GPU instead of replacing nodes
+    """
+    def add_requirements(self, fgraph):
+        fgraph.attach_feature(toolbox.ReplaceValidate())
+
+    def apply(self, fgraph):
+        mapping = {}
+        move_to_GPU = True
+
+        # Building a new graph
+        for i in fgraph.inputs:
+            mapping[i] = GpuFromHost(None)(i)
```

Comment:
And since an input of a graph should always be an input of an apply node, this won't cause a problem while doing `replace_validate` at the end. Am I correct ?